tests/sanic/test_sanic.py

```diff
@@ -77,28 +93,20 @@ def dockerflow_redis(app):
 
 @pytest.fixture
 def test_client(app):
-    # Create SanicTestClient manually and provide a socket object instead of host
-    # and port when calling Sanic.run in order to avoid parallel test failures
-    # caused by Sanic.test_client bindngs to a static port
-    s = socket.socket()
-    s.bind((sanic.testing.HOST, 0))
-    try:
-        # initialize test_client with socket's port
-        test_client = sanic.testing.SanicTestClient(app, s.getsockname()[1])
-        # override app.run to drop host and port in favor of socket
-        run = app.run
-        app.run = lambda host, port, **kw: run(sock=s, **kw)
-        # yield test_client
-        yield test_client
-    finally:
-        s.close()
+    return SanicTestClient(app)
 
 
 def test_instantiating(app):
-    dockerflow = Dockerflow()
-    assert "dockerflow.heartbeat" not in app.router.routes_names
-    dockerflow.init_app(app)
-    assert "dockerflow.heartbeat" in app.router.routes_names
+    Dockerflow()
+    if sanic.__version__.startswith("20."):
```

Comment:
Nit: I prefer to use `@pytest.skipif` and two separate tests. That way, when we drop old versions, we delete a whole test rather than lines in a test.

```python
@pytest.skipif(not sanic.__version__.startswith("20."), "requires sanic 20")
def test_instantiating_sanic_20(app):
    dockerflow = Dockerflow()
    assert "dockerflow.heartbeat" not in app.router.routes_names
    dockerflow.init_app(app)
    assert "dockerflow.heartbeat" in app.router.routes_names
    
@pytest.skipif(sanic.__version__.startswith("20."), "requires sanic 21 or later")
def test_instantiating(app):
    Dockerflow()
    assert ("__heartbeat__",) not in app.router.routes_all
    Dockerflow(app)
    assert ("__heartbeat__",) in app.router.routes_all
```